pkg/lib/userexport/model.go

```diff
@@ -0,0 +1,121 @@
+
+	Identities     []*identity.Info
+	Authenticators []*authenticator.Info
+}
+
+type Error struct {
+	Message string `json:"message,omitempty"`
+	Reason  string `json:"reason,omitempty"`
+}
+
+var RequestSchema = validation.NewSimpleSchema(`
+{
+	"type": "object",
+	"additionalProperties": false,
+	"properties": {
+		"format": {
+			"type": "string"
+		}
+	},
+	"required": ["format"]
+}
+`)
+
+type Request struct {
+	Format string `json:"format"`
+}
+
+type Response struct {
+	ID          string                `json:"id,omitempty"`
+	CreatedAt   *time.Time            `json:"created_at,omitempty"`
+	CompletedAt *time.Time            `json:"completed_at,omitempty"`
+	FailedAt    *time.Time            `json:"failed_at,omitempty"`
+	Status      redisqueue.TaskStatus `json:"status,omitempty"`
+	Request     *Request              `json:"request,omitempty"`
+	DownloadUrl string                `json:"download_url,omitempty"`
+	Error       *Error                `json:"error,omitempty"`
+}
+
+type Address struct {
+	Formatted     string `json:"formatted"`
+	StreetAddress string `json:"street_address"`
+	Locality      string `json:"locality"`
+	Region        string `json:"region"`
+	PostalCode    string `json:"postal_code"`
+	Country       string `json:"country"`
+}
+
+type Identity struct {
+	Type    model.IdentityType     `json:"type"`
+	LoginID map[string]interface{} `json:"login_id,omitempty"`
+	OAuth   map[string]interface{} `json:"oauth,omitempty"`
+	LDAP    map[string]interface{} `json:"ldap,omitempty"`
+	Claims  map[string]interface{} `json:"claims,omitempty"`
+}
+
+type MFATOTP struct {
+	Secret string `json:"secret"`
+	URI    string `json:"uri"`
+}
+
+type MFA struct {
+	Emails       []string   `json:"emails"`
+	PhoneNumbers []string   `json:"phone_numbers"`
+	TOTPs        []*MFATOTP `json:"totps"`
+}
+
+type Record struct {
+	Sub string `json:"sub"`
+
+	PreferredUsername string `json:"preferred_username"`
+	Email             string `json:"email"`
+	PhoneNumber       string `json:"phone_number"`
+
+	EmailVerified       bool `json:"email_verified"`
+	PhoneNumberVerified bool `json:"phone_number_verified"`
+
+	Name       string   `json:"name"`
+	GivenName  string   `json:"given_name"`
+	FamilyName string   `json:"family_name"`
+	MiddleName string   `json:"middle_name"`
+	Nickname   string   `json:"nickname"`
+	Profile    string   `json:"profile"`
+	Picture    string   `json:"picture"`
+	Website    string   `json:"website"`
+	Gender     string   `json:"gender"`
+	Birthdate  string   `json:"birthdate"`
+	Zoneinfo   string   `json:"zoneinfo"`
+	Locale     string   `json:"locale"`
+	Address    *Address `json:"address"`
+
+	CustomAttributes map[string]interface{} `json:"custom_attributes"`
+
+	Roles  []string `json:"roles"`
+	Groups []string `json:"groups"`
+
+	Disabled bool `json:"disabled"`
+
+	Identities []*Identity `json:"identities"`
+
+	Mfa *MFA `json:"mfa"`
+
+	BiometricCount int `json:"biometric_count"`
+	PasskeyCount   int `json:"passkey_count"`
```

Comment:
For strings, add omitempty.
For maps, add omitempty.
For slices, add omitempty.
For bools, DO NOT add omitempty. (That is, you make no changes to them. They are correct now)

Reply:
added omitempty